### PR TITLE
[#1124] docs: Add the document for tez-client

### PR DESCRIPTION
**Pull request by bin41215**
### What changes were proposed in this pull request?

Add the document for tez-client.

### Why are the changes needed?

(Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.)

Fix: #1124 

### Does this PR introduce _any_ user-facing change?
No.

### CODE REVIEW

File: docs/client_guide.md [hunk 1]
```diff
@@ -241,7 +241,8 @@ Notice: this feature requires the MEMORY_LOCAL_HADOOP mode.
 | Property Name                  | Default | Description                                                             |
 |--------------------------------|---------|-------------------------------------------------------------------------|
 | tez.rss.avoid.recompute.succeeded.task | false   | Whether to avoid recompute succeeded task when node is unhealthy or black-listed |
- 
+| tez.rss.client.max.buffer.size | 3k | The max buffer size in map side |
```
Comment (jerqi) [inline docs/client_guide.md:244]: What's the difference between this option and `rss.writer.buffer.size`? Could you elaborate this description?

Comment (bin41215) [reply docs/client_guide.md:244]: ok.

Comment (bin41215) [reply docs/client_guide.md:244]: rss.client.max.buffer.size  -->  maxSegmentSize -->  This parameter is the size of the WrappedBuffer created
rss.writer.buffer.size      -->  maxBufferSize  -->  When buffer.getDataLength > maxBufferSize, the buffer will be sent to the server
This logic is a reuse of MapReduce.

**Review by jerqi — APPROVED** [review]
LGTM, merged to master.

**Pull request merged**